### PR TITLE
Initial changes to support testing

### PR DESCRIPTION
**Pull request by c-simpson**
As per Issue #146.   Includes: 
- suggested fix for Issue #147  (404 from /api/statistics)
- Addition of pytest to requirements.txt
- Initial DNS and simple API tests

Sorry about the re-formatting. Can revert and fix if a hassle.

### CODE REVIEW

File: src/server/api/admin_api.py [hunk 1]
```diff
@@ -99,33 +108,35 @@ def checkStatus():
...
 def listStatistics():
     try:
-        last_execution_file = open(LOGS_PATH + 'last_execution.json', 'r')
+        last_execution_file = open(LOGS_PATH + "last_execution.json", "r")
         last_execution_details = json.loads(last_execution_file.read())
         last_execution_file.close()
 
     except Exception as e:
-        return abort(404)
+        return jsonify(None)
```
Comment (dkelley) [inline src/server/api/admin_api.py:122]: What if we have a real exception?  Can't we do a simple file test rather than catch and ignore the exception?

Comment (c-simpson) [reply src/server/api/admin_api.py:122]: I have no problem with that idea but I'm not sure how we'd handle anything else. If the JSON was bad and we got a JSONDecodeError we'd probably want to log it but we couldn't pass it on.

Comment (dkelley) [reply src/server/api/admin_api.py:122]: Correct, that should be logged and a 500 error returned so you know there is a server error.  I'm not sure how flask handles exceptions bubbling up, but worth a look

Comment (c-simpson) [reply src/server/api/admin_api.py]: I'm now logging and returning a 500 if the file is missing or not valid JSON. I've tested each case.
I'm also creating a minimal valid (null) last_execution.json file if there's not one already, so if we get an error here it's a real problem. 
I'll create an issue to notify the client folks about the possibility of a 500.

**Pull request merged**